core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java

```diff
@@ -1570,6 +1570,90 @@ public void pow_null_missing() {
     assertTrue(power.valueOf(valueEnv()).isMissing());
   }
 
+  /**
+   * Test rint with byte value.
+   */
+  @ParameterizedTest(name = "rint({0})")
+  @ValueSource(bytes = {
+      -1, 0, 1, Byte.MAX_VALUE, Byte.MIN_VALUE})
+  public void rint_byte_value(Byte value) {
+    FunctionExpression rint = DSL.rint(DSL.literal(value));
+    assertThat(
+            rint.valueOf(valueEnv()),
+            allOf(hasType(DOUBLE), hasValue(Math.rint(value))));
+    assertEquals(String.format("rint(%s)", value), rint.toString());
+  }
+
+  /**
+   * Test rint with short value.
+   */
+  @ParameterizedTest(name = "rint({0})")
+  @ValueSource(shorts = {
+      -1, 0, 1, Short.MAX_VALUE, Short.MIN_VALUE})
+  public void rint_short_value(Short value) {
+    FunctionExpression rint = DSL.rint(DSL.literal(value));
+    assertThat(
+            rint.valueOf(valueEnv()),
+            allOf(hasType(DOUBLE), hasValue(Math.rint(value))));
+    assertEquals(String.format("rint(%s)", value), rint.toString());
+  }
+
+  /**
+   * Test rint with integer value.
+   */
+  @ParameterizedTest(name = "rint({0})")
+  @ValueSource(ints = {
+      -1, 0, 1, Integer.MAX_VALUE, Integer.MIN_VALUE})
+  public void rint_int_value(Integer value) {
+    FunctionExpression rint = DSL.rint(DSL.literal(value));
+    assertThat(
+            rint.valueOf(valueEnv()),
+            allOf(hasType(DOUBLE), hasValue(Math.rint(value))));
+    assertEquals(String.format("rint(%s)", value), rint.toString());
+  }
+
+  /**
+   * Test rint with long value.
+   */
+  @ParameterizedTest(name = "rint({0})")
+  @ValueSource(longs = {
+      -1L, 0L, 1L, Long.MAX_VALUE, Long.MIN_VALUE})
+  public void rint_long_value(Long value) {
+    FunctionExpression rint = DSL.rint(DSL.literal(value));
+    assertThat(
+            rint.valueOf(valueEnv()),
+            allOf(hasType(DOUBLE), hasValue(Math.rint(value))));
+    assertEquals(String.format("rint(%s)", value), rint.toString());
+  }
+
+  /**
+   * Test rint with float value.
+   */
+  @ParameterizedTest(name = "rint({0})")
+  @ValueSource(floats = {
+      -1F, -0.5F, 0F, 0.5F, 1F, Float.MAX_VALUE, Float.MIN_VALUE})
+  public void rint_float_value(Float value) {
+    FunctionExpression rint = DSL.rint(DSL.literal(value));
+    assertThat(
+            rint.valueOf(valueEnv()),
+            allOf(hasType(DOUBLE), hasValue(Math.rint(value))));
+    assertEquals(String.format("rint(%s)", value), rint.toString());
+  }
+
+  /**
+   * Test rint with double value.
+   */
+  @ParameterizedTest(name = "rint({0})")
+  @ValueSource(doubles = {
+      -1D, -0.5D, 0D, 0.5D, 1D, Double.MAX_VALUE, Double.MIN_VALUE})
```

Comment:
Maybe try more values, like 0.500000023, 42.42 and so on

Reply:
fixed in `95e935e`